src/app/tests/TestAttributeValueEncoder.cpp

```diff
@@ -238,6 +239,59 @@ void TestEncodeEmptyList(nlTestSuite * aSuite, void * aContext)
     VERIFY_BUFFER_STATE(aSuite, test, expected);
 }
 
+void TestEncodeFabricScoped(nlTestSuite * aSuite, void * aContext)
+{
+    TestSetup test(aSuite, 1 /* Fabric Index */);
+    Clusters::AccessControl::Structs::ExtensionEntry::Type items[3];
+    items[0].fabricIndex = 0;
+    items[1].fabricIndex = 1;
+    items[2].fabricIndex = 2;
+
+    // We tried to encode three items, however, the encoder should only put the item with matching fabric index into the final list.
+    CHIP_ERROR err = test.encoder.EncodeList([items](const auto & encoder) -> CHIP_ERROR {
+        for (size_t i = 0; i < 3; i++)
+        {
+            ReturnErrorOnFailure(encoder.Encode(items[i]));
+        }
+        return CHIP_NO_ERROR;
+    });
+    NL_TEST_ASSERT(aSuite, err == CHIP_NO_ERROR);
+    const uint8_t expected[] = {
```

Comment:
Can we please avoid these raw, TLV byte strings? Instead, decoding the value back should result in just the one item with the matching fabric index. That would achieve equivalent testing coverage.

Reply:
It could be a bit hard to do so since we are not in the context of read interactions. (However, this test should be in this file since the changes are for the attribute value encoder)

Added a TODO in the file for refactroing the whold unit test.